src/main/scala/no/ndla/searchapi/integration/GrepApiClient.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Part of NDLA search-api.
+ * Copyright (C) 2020 NDLA
+ *
+ * See LICENSE
+ */
+
+package no.ndla.searchapi.integration
+
+import java.util.concurrent.Executors
+
+import com.typesafe.scalalogging.LazyLogging
+import no.ndla.network.NdlaClient
+import no.ndla.searchapi.SearchApiProperties.GrepApiUrl
+import no.ndla.searchapi.model.api.GrepException
+import no.ndla.searchapi.model.domain.RequestInfo
+import no.ndla.searchapi.model.grep._
+import org.json4s.DefaultFormats
+import scalaj.http.Http
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, ExecutionContextExecutor, Future}
+import scala.util.{Failure, Success, Try}
+
+trait GrepApiClient {
+  this: NdlaClient =>
+  val grepApiClient: GrepApiClient
+
+  class GrepApiClient extends LazyLogging {
+    implicit val formats: DefaultFormats.type = org.json4s.DefaultFormats
+    private val GrepApiEndpoint = s"$GrepApiUrl/kl06/v201906"
+
+    def getAllKjerneelementer: Try[List[GrepElement]] =
+      get[List[GrepElement]](s"$GrepApiEndpoint/kjerneelementer-lk20/").map(_.distinct)
+
+    def getAllKompetansemaal: Try[List[GrepElement]] =
+      get[List[GrepElement]](s"$GrepApiEndpoint/kompetansemaal-lk20/").map(_.distinct)
+
+    def getAllTverrfagligeTemaer: Try[List[GrepElement]] =
+      get[List[GrepElement]](s"$GrepApiEndpoint/tverrfaglige-temaer-lk20/").map(_.distinct)
+
+    def getGrepBundle: Try[GrepBundle] = {
+      logger.info("Fetching grep in bulk...")
+      val startFetch = System.currentTimeMillis()
+      implicit val ec: ExecutionContextExecutor = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(3))
+
+      val requestInfo = RequestInfo()
+
+      /** Calls function in separate thread and converts Try to Future */
+      def tryToFuture[T](x: () => Try[T]) = Future { requestInfo.setRequestInfo(); x() }.flatMap(Future.fromTry)
+
+      val kjerneelementer = tryToFuture(getAllKjerneelementer _)
+      val kompetansemaal = tryToFuture(getAllKompetansemaal _)
+      val tverrfagligeTemaer = tryToFuture(getAllTverrfagligeTemaer _)
+
+      val x = for {
+        f1 <- kjerneelementer
+        f2 <- kompetansemaal
+        f3 <- tverrfagligeTemaer
+      } yield GrepBundle(f1, f2, f3)
+
+      Try(Await.result(x, Duration(300, "seconds"))) match {
+        case Success(bundle) =>
+          logger.info(s"Fetched grep in ${System.currentTimeMillis() - startFetch}ms...")
+          println(Success(bundle))
+          Success(bundle)
+        case Failure(ex) =>
+          logger.error(s"Could not fetch grep bundle (${ex.getMessage})", ex)
+          Failure(GrepException("Could not fetch grep bundle..."))
+      }
+    }
+
+    private def get[A](url: String, params: (String, String)*)(implicit mf: Manifest[A]): Try[A] = {
+      ndlaClient.fetchWithForwardedAuth[A](Http(url).timeout(60000, 60000).params(params))
```

Comment:
Her burde vi vel unngå å forwarde authen til det eksterne apiet?

Reply:
Enig. Grep har ikkje autentisering så det gir ikkje meining å sende den videre.

Reply:
var litt for ivrig med kopieringa fra taxonomy, fixing now